### PR TITLE
Upgrade netty-codec version to 4.1.86.Final

### PR DESCRIPTION
**Pull request by schoims**
netty-codec (v.4.1.77.Final) doesn't have ValueValidator in DefaultHeaders class, so the version needs to be upgraded to 4.1.86.Final.

error message from PR (https://github.com/Azure/azure-iot-sdk-java/pull/1645)
image

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -124,12 +124,12 @@
             <dependency>
                 <groupId>io.netty</groupId>
                 <artifactId>netty-codec</artifactId>
-                <version>4.1.77.Final</version> <!--e2e tests and samples only-->
+                <version>4.1.86.Final</version> <!--e2e tests and samples only-->
             </dependency>
             <dependency>
                 <groupId>io.netty</groupId>
                 <artifactId>netty-codec-http</artifactId>
-                <version>4.1.77.Final</version> <!--e2e tests and samples only-->
```
Comment (abhipsaMisra) [inline pom.xml:132]: Do we need to update "netty-codec-http2" and "reactor-netty-http" versions as well?

Comment (schoims) [reply pom.xml:132]: It seems like the gate is passing without upgrading the versions of those packages.

**Review by patilsnr — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (abhipsaMisra) [reply pom.xml:132]: I'd update "netty-codec-http2", just so it follows the "netty-codec-http" version. (netty-codec-http2 has a dependency on netty-codec-http, so it should get the vulnerability patch automatically).

@timtay-microsoft - is the http2 lib required for using HTTP2.0?

Comment (schoims) [reply pom.xml:132]: Ah sure! Thanks for checking!

Comment (timtay-microsoft) [reply pom.xml:132]: Just upgrading the vulnerable package is fine here

Comment (timtay-microsoft) [reply pom.xml:132]: Upgrading netty packages has caused some issues with our Azure Core dependencies in the past, so upgrading everything may cause some unusual test failures

**Review by timtay-microsoft — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (schoims) [reply pom.xml:132]: I upgraded the two packages and the gate passed. Am I good to merge it in?

Comment (timtay-microsoft) [reply pom.xml:132]: Yeah, go ahead

**Pull request merged**